Documentation/Signoffs/Microcontroller.md

```diff
@@ -0,0 +1,44 @@
+# Microcontroller Signoff
+## FUNCTION OF THE SUBSYSTEM
+Within this subsystem, the code will instruct the hardware in how to sample from the sensors, send data to the Communication subsystem, format and store data onto a USB Flash drive, and monitor battery voltage and current levels from the power subsystem.  These are the core functions that are desired from the microcontroller.  
+
+## CONSTRAINTS ON THE SUBSYSTEM
+The constraints given below are generalized to allow for comparison of any potential contender for the microcontroller.  Constraints on the subsystem are as follows:
+1. The chosen microcontroller should cost no more than $60 as the end-user only has a budget of $300.  This board value cap is gathered from estimating the costs of other subsystems and viewing prices of available options that satisfy proceeding constraints such as Raspberry Pis, Arduinos, Intel, ESP, etc.  Up to 20% of the budget was then allocated due to the importance of this subsystem.
+2. The microcontroller must be able to send instructions and transfer data to other components such as the Communication subsystem.  To do this, Serial Peripheral Interface (SPI) or Universal Asynchronous Receiver/Transmitter (UART) (2 extra UART pins) as standards in hardware communication for liaising will need to be achievable. 
+3. USB Flash Drive external storage support is required as per the shall statements. The USB standard is used for convenience for the user as it is small, easily portable, commonly used for data storage, and readily available.
+4. The board must have at least 15 digital input pins and 16 analog input pins. The board must have enough Input/Output (I/O) pins to operate any combination of 3 sensors whether they be digital or analog. Different sensors interface with a potentially different number of pins; preliminary research shows most air quality sensors take 5 of fewer pins to operate, so 5 pins per sensor times 3 sensors gives 15 pins. 2 analog pins will be used to measure voltage and current from the power subsystem to monitor battery capacity.  If using UART, 2 UART pins will also be needed.  If using SPI, 3 extra digital pins are needed.
+5. There must be sufficient memory for the code. Comparing similar operating code (Wireless Sensor Collection and Small-time Process Operating) with libraries and compilation show a range of 15 KB (10%) (Kilobytes) above or below 150 KB in flash memory.  To account for the worst case, 165 KB is the required memory.
+6. Some form of battery monitoring capabilities need to be included or applicable as per shall statements.
```

Comment:
"Some form of..." is too vague.

Reply:
This constraint has be moved to the Power subsystem after thorough consideration. There is still some remains of it here as the microcontroller will still need to take in some kind of data from the Power subsystem to tell battery capacity.